app/src/main/java/com/mediaan/mymediaan/util/SharedPrefsUtil.kt

```diff
@@ -0,0 +1,21 @@
+package com.mediaan.mymediaan.util
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class SharedPrefsUtil {
+    private val SHARED_PREFS_NAME = "MyMediaanAppPrefs"
+    private val LOGGED_IN_ID_KEY = "logged_in_user"
+
+    fun saveLoggedInUserId(context: Context, id: String) {
+        val sharedPreferences: SharedPreferences = context.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE)
+        val editor = sharedPreferences.edit()
+        editor.putString(LOGGED_IN_ID_KEY, id)
+        editor.apply()
+    }
+
+    fun getLoggedInUserId(context: Context): String {
+        val sharedPreferences: SharedPreferences = context.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE)
+        return sharedPreferences.getString(LOGGED_IN_ID_KEY, "") ?: ""
+    }
+}
```

Comment:
While [reading up on SharedPreferences API](https://developer.android.com/training/data-storage/shared-preferences), I noticed that it's not a recommended approach anymore:

> Caution: DataStore is a modern data storage solution that you should use instead of SharedPreferences. It builds on Kotlin coroutines and Flow, and overcomes many of the drawbacks of SharedPreferences.

Do you feel like [upgrading code to DataStore](https://developer.android.com/topic/libraries/architecture/datastore)?

Reply:
I know about it, but I chose to go for sharedpreferences because it is a tiny bit more performant for our use case(since DataStore uses Kotlin coroutines and Flow to store data asynchronously, consistently, and transactionally). This way we get the id synchronously without any added effort, and all we do is store one string. I can refactor it, if you insist on using a DataStore. I don't feel too strongly about it.